### PR TITLE
added cards on home screen

### PR DESCRIPTION
**Pull request by Akhil2105**
## Summary
- added cards + cart items
- click on cart takes you to items added

### CODE REVIEW

File: src/app/cart.service.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  private cartItemSubject = new BehaviorSubject<any[]>([]);
+  cartItem$ = this.cartItemSubject.asObservable();
+
+  addToCart(item: any): void {
+    const currentCart = this.cartItemSubject.value;
+    const updatedCart = [...currentCart, item];
+    this.cartItemSubject.next(updatedCart);
+  }
+
+  getCartItems(): any[] {
```
Comment (ng-model) [inline src/app/cart.service.ts:17]: // use proper return type here

File: src/app/cart/cart.component.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<div *ngIf="cartItems.length > 0">
+    <h2 class="product-details">Shopping Cart</h2>
+    <ul>
+        <li *ngFor="let item of cartItems">
+            <h3>{{item.name}}</h3>
+            <p>Price: {{item.price | currency:'USD':'symbol'}}
+        </li>
+    </ul>
+</div>
+<div *ngIf="cartItems.length === 0">
```
Comment (ng-model) [inline src/app/cart/cart.component.html:10]: use else condition.

File: src/app/cart/cart.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { CartService } from './../cart.service';
+import { Component, OnInit } from '@angular/core';
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styleUrls: ['./cart.component.scss']
+})
+export class CartComponent implements OnInit {
+  cartItems: any[] = [];
```
Comment (ng-model) [inline src/app/cart/cart.component.ts:9]: use proper types

File: src/app/cart/cart.component.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  constructor(private cartService: CartService) { }
+
+  ngOnInit(): void {
+    this.cartService.cartItem$.subscribe(items => {
+      this.cartItems = items;
+      console.log('Cart Items:', this.cartItems);
```
Comment (ng-model) [inline src/app/cart/cart.component.ts:15]: no logs please

File: src/app/home/home.component.ts [hunk 5]
```diff
@@ -54,5 +39,21 @@ export class HomeComponent implements OnInit {
     { "name": "SoundWave Freedom", "price": 69.99, "category": "Electronics", "description": "Enjoy wire-free audio with these stylish and high-quality Bluetooth earbuds." }
   ];
```
Comment (ng-model) [inline src/app/home/home.component.ts:40]: move mock data to a json file

File: src/app/home/home.component.ts [hunk 6]
```diff
@@ -54,5 +39,21 @@ export class HomeComponent implements OnInit {
...
 
+  ngOnInit(): void {
+    this.All = this.products.length;
+    this.household = this.products.filter(product => product.category === 'Household').length;
+    this.mobile = this.products.filter(product => product.category === 'Mobile').length;
+    this.accs = this.products.filter(product => product.category === 'Accessories').length;
```
Comment (ng-model) [inline src/app/home/home.component.ts:46]: re-use

**Review by ng-model — APPROVED** [review]

**Comment by Akhil2105** [comment]
will work on the issues.

**Pull request merged**